puppet/chassis.rb

```diff
@@ -177,7 +177,25 @@ def self.install_extension(extension)
 			repo = extension
 		end
 
-		folder = @@dir + '/extensions/' + extension.split('/').last.gsub(/.git$/, '').downcase
+		# We need to add a version to allow for a seamless upgrade to Bionic Beaver.
+		if config["version"] >= 3
+			# Check for existing extensions that have a hyphen and remove them.
+			old_extensions = Dir.glob(@@dir + '/extensions/*')
+			old_extensions.each { |extension|
+				next unless extension.include? "-"
```

Comment:
This isn't a block now, so I think we can deindent the next lines (and remove `end`)?

Reply:
Done!